### 4.1（1） 人体出汗数据：单个总体的均值检验(Σ未知)

In [145]:
# 总体均值检验

library(readxl)
data = read_excel('.//AMSA_data//《应用多元统计分析》（第6版）Excel数据//exec4.1.xlsx')
head(data) #查看前几行

出汗量 钠含量 钾含量
1 3.7    48.5    9.3  
2 5.7    65.1    8.0  
3 3.8    47.2   10.9  
4 3.2    53.2   12.0  
5 3.1    55.5    9.7  
6 4.6    36.1    7.9

In [146]:
n = dim(data)[1] # 数据长度
p = 3 # 维度
mu0 = c(4,50,10)
meanx = apply(data,2,mean)
data.frame(meanx) # 样本均值

meanx 
出汗量  4.640
钠含量 45.400
钾含量  9.965

In [147]:
Ts = n * t(meanx-mu0) %*% solve(cov(data)) %*% (meanx-mu0) # 计算T方统计量
F = (n-p)/p/(n-1)*Ts
p = 1-pf(F,p,n-p) 
p # 计算p值 

[,1]      
[1,] 0.06492834

所以，在显著性水平α=0.05下，没有理由拒绝原假设，即认为健康成年女性的出汗量、钠含量、钾含量的均值与（5，50，10）没有显著差异。

### 4.3 轮胎测试数据：两个总体均值的比较分析（Σ未知但是相等）

In [1]:
library(readxl)
data = read_excel('.//AMSA_data//《应用多元统计分析》（第6版）Excel数据//exec4.3.xlsx')
head(data) #查看前几行

第一阶段耐用性指标值 第二阶段耐用性指标值 第三阶段耐用性指标值 品牌
1 194                  192                  141                  甲  
2 208                  188                  165                  甲  
3 233                  217                  171                  甲  
4 241                  222                  201                  甲  
5 265                  252                  207                  甲  
6 269                  283                  191                  甲

In [151]:
data1 = subset(data, 品牌=='甲')[,1:3]
data2 = subset(data, 品牌=='乙')[,1:3]
n = dim(data1)[1]  # dim(data1)[1] == dim(data2)[1] TRUE
p = 3
meanx = apply(data1,2,mean)
meany = apply(data2,2,mean)
data.frame(meanx)
data.frame(meany)

meanx   
第一阶段耐用性指标值 235.0000
第二阶段耐用性指标值 225.6667
第三阶段耐用性指标值 179.3333

meany    
第一阶段耐用性指标值 227.33333
第二阶段耐用性指标值 120.00000
第三阶段耐用性指标值  89.83333

In [152]:
Si = function(data,mean){
    sum = 0
    for(i in 1:n){
        sum = sum + (t(as.matrix(data[i,]-mean))) %*%  (as.matrix(data[i,]-mean))
    }
    S = (1/(n-1))*sum
    return(S)
}

S1 = Si(data1,meanx)
S2 = Si(data2,meany)
Sp = ((n-1)*S1+(n-1)*S2)/(n+n-2)

A = as.matrix(meanx-meany)  ;B= t(A)
Ts = (n*n/(n+n)) * B %*% solve(Sp) %*% A
F = (n+n-p-1)/(p*(n+n-2))*Ts
pv = 1-pf(F,p,n+n-p-1) 
pv # 计算p值

[,1]       
[1,] 1.22699e-06

在显著性水平α=0.05下，可以拒绝原假设，即认为甲乙两种品牌轮胎的耐用性指标有显著差异。

In [153]:
as.matrix(meanx-meany) # 计算均值差值

[,1]      
第一阶段耐用性指标值   7.666667
第二阶段耐用性指标值 105.666667
第三阶段耐用性指标值  89.500000

显然，甲乙两种产品在第二个阶段的锻造过程中存在明显的差异，第三阶段次之，而在第一阶段则不存在较大差异。

还可以使用多元方差分析的方法从统计上观察差异的差异性。

In [2]:
library(MASS)
attach(data)
g = factor(品牌)  # 转换为因子
y = cbind(第一阶段耐用性指标值,第二阶段耐用性指标值,第三阶段耐用性指标值)
detach(data)
fit = manova(y~g) # 多元方差分析
summary.aov(fit) # 一元方差分析表

 Response 第一阶段耐用性指标值 :
            Df Sum Sq Mean Sq F value Pr(>F)
g            1  176.3  176.33  0.2666 0.6169
Residuals   10 6615.3  661.53               

 Response 第二阶段耐用性指标值 :
            Df Sum Sq Mean Sq F value    Pr(>F)    
g            1  33496   33496  48.215 3.978e-05 ***
Residuals   10   6947     695                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

 Response 第三阶段耐用性指标值 :
            Df  Sum Sq Mean Sq F value    Pr(>F)    
g            1 24030.8 24030.8  60.285 1.528e-05 ***
Residuals   10  3986.2   398.6                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1


分析三对数据方差分析的p值，对于第一阶段数据，P值为0.6169，没有理由拒绝原假设，在0.05的显著性水平下认为甲乙两个品牌在受第一阶段影响差异并不显著。同理，第二阶段和第三阶段方差分析的P值结果都远远小于0.05，即认为甲乙两个品牌在受第二、三阶段影响差异非常显著。这和我们直观观察得到的结论也是相同的。